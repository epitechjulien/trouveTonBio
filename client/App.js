import React from 'react';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import ReduxThunk from 'redux-thunk';
import  { init } from './helpers/db';
import PlacesNavigator from './navigation/PlacesNavigator';
import placesReducer from './store/places-reducer';

init()
  .then(() => {
    console.log('Initiliazed database');
})
.catch(err => {
  console.log('Initializing db failed.');
  console.log(err);
});

const rootReducer = combineReducers({
  places: placesReducer
});

const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

export default function App() {
  return (
    <Provider store={store}>
      <PlacesNavigator />
    </Provider>
  );
}
