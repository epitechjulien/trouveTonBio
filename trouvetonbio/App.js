import React, {useState} from 'react';
import { Text, View } from 'react-native';
import * as Font from 'expo-font';
import { AppLoading } from 'expo';
import { useScreens } from 'react-native-screens';

// STORE import
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import cartReducer from './store/reducers/cart';
import productsReducer from './store/reducers/products';
import ShopNavigator from './navigation/ShopNavigator';
import ordersReducer from './store/reducers/orders';

// Product import
import ProductsNavigator from './navigation/ProductsNavigator';

useScreens();

// Font loading import
const fetchFonts = () => {
  Font.loadAsync({
    'open-sans': require('./assets/fonts/OpenSans-Regular.ttf'),
    'open-sans-bold': require('./assets/fonts/OpenSans-Bold.ttf')
  });
};

// Reducer const combiner
const rootReducer = combineReducers({
  products: productsReducer,
  cart: cartReducer,
  orders: ordersReducer
});

// remove composeWithDev before deploying app
const store = createStore(rootReducer);

export default function App() {
  const [fontLoaded, setFontLoaded] = useState(false);

  if(!fontLoaded) {
    return (
      <AppLoading
        startAsync={fetchFonts}
        onFinish={() => setFontLoaded(true)}
      />
    );
  }

  return <ProductsNavigator />;
}
