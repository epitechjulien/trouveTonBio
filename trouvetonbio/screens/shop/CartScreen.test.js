import React from 'react';
import renderer from 'react-test-renderer';
import App from "../../App"
import CartScreen from "./CartScreen"
import { Provider } from 'react-redux'
import { createStore } from 'redux'

it('renders correctly', () => {
  const tree = renderer
  .create(
    <Provider store={ createStore(state => { return state }, { cart: { totalAmount: 0 } }) } >
      <CartScreen />
    </Provider>
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});



// test('one is one', () => {
//   expect(1).toEqual(1)
// });