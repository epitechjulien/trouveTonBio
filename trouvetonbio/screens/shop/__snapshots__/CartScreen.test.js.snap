// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "padding": 10,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.26,
        "shadowRadius": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "open-sans-bold",
          "fontSize": 18,
        }
      }
    >
      Total:
      <Text
        style={
          Object {
            "color": "#368654",
          }
        }
      >
        $
        0.00
      </Text>
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={
        Array [
          "disabled",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#368654",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Order Now
        </Text>
      </View>
    </View>
  </View>
  <View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
